Widget and configurable upload server

cooperq opened the pull request:
Hey man, these commits include configurable endpoints and a widget for one touch recording.  Talk to me on irc or email if you have any questions / comments.

Hainish committed:
Removing binaries

Hainish committed:
Adding derived files to .gitignore

cooperq committed:
adding support for user configurable upload server, default is openwatch.net

cooperq committed:
Merge branch 'master' of github.com:cooperq/OpenWatch---Android

cooperq committed:
adding support for user configurable upload server, default is openwatch.net

cooperq committed:
adding r.java file back into repo

cooperq committed:
adding r.java file back into repo

Hainish committed:
Creating a widget for automatic video recording

cooperq committed:
fixing merge

cooperq committed:
putting r.java back

Miserlou commented:
THIS IS BADASS.

I'm going to have to take some time to review these changes as soon as I can but.. yeah. Awesome stuff.

cooperq committed:
fixing weird layout bug

cooperq committed:
adding properties file

Miserlou commented:
Hey @cooperq and @Hainish - can you guys join the Google group to discuss further development of this? http://groups.google.com/group/the-openwatch-project?lnk=srg

Commit:
This fixes issues where uploadService.java and recordService.java won't
automatically generate.

cooperq committed:
removing generated files from repo

cooperq committed:
adding tasks to TODO list

Commit:
added need for code comments

Commit:
Added a bunch of comments so shit makes sense. Also added some todo
items to de-mystify some sections of code and make efficieny
improvements.

Commit:
minor comments

Commit:
Removed some imports which weren't being used. Should lower memory
footprint a little and remove some warnings from eclipse.

Commit:
Added legal info for first-run disclaimer.

Commit:
Fixed typo

hbringo committed:
updated legal info todo item

hbringo committed:
Added items to preference menu for user to select custom upload path, whether the server requires ssl, and what port to use.

hbringo committed:
Added support for custom upload path. Up next: custom port and proxy support!

Commit:
added ability to upload to custom path still debugging some issues on
this

Commit:
Finally got toast to display, looks like routine is working. Need to
test on live upload server.

cooperq committed:
fixing upload path

hbringo committed:
added some questions

hbringo committed:
OpenWatch will now notify users when an upload fails. Tested and works. We need to make this a status bar notification instead of a toast if possible.

Commit:
worked a little on non-overt recording mode

Commit:
App now notifies users if there are problems using the save directory,
the SD card is not mounted, or there are problems stopping the video
recording. These should eventually be switched to notifications instead
of toasts.

Commit:
Working on video previewer

Commit:
Added toast notifications for auto-generated catch blocks. Should switch
to status bar notifications if possible.

Commit:
Cleaning up some TODOs we didn't need.

Commit:
Added an item to the preferences menu so users can choose overt or
covert video recording mode

Commit:
Implemented feature on todo list for an overt video recording mode.
Screen now shows a preview of what it's recording. Added need for overt
audio recording mode and post-record video/audio preview to the todo
list. Created a new layout view for overt recording.

Commit:
Changed disclaimer and implemented feature request for a "tell me more"
option on it. Sends users to the Reporters Committee for the Freedom
of the Press guide called "Can we Tape?" at
http://www.rcfp.org/can-we-tape.

Commit:
Working on auto-preview option

Commit:
more working on this

Commit:
fixed force close errors

Commit:
moved cancel to negativebutton

Commit:
now actually goes to upload screen when selected

Commit:
Implemented feature request so users can now preview videos directly
after recording. This could still use some usability improvements see
the updated TODO list. Also made language more clear on preferences menu
re: what features are actually implemented right now.
